### PR TITLE
Fixes issue #591: delete method regression

### PR DESCRIPTION
**Pull request by mattyr**

### CODE REVIEW

File: spec/issues/591_delete_method_regression_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    JSON.parse(last_response.body)
+  end
+
+  specify do
+    expect(subject['paths'].map { |_, p| p['delete']['responses'] }).to eql(
+      [
+        { '200' => { 'description' => 'deleted', 'schema' => { '$ref' => '#/definitions/Foo' } } },
+        { '200' => { 'description' => 'deleted', 'schema' => { '$ref' => '#/definitions/Foo' } } },
+        { '203' => { 'description' => 'deleted' }, '200' => { 'schema' => { '$ref' => '#/definitions/Foo' } } }
+      ]
```
Comment (LeFnord) [inline spec/issues/591_delete_method_regression_spec.rb]: a bit curious, you have 3 endpoints → routes, but 4 success responses

File: spec/issues/591_delete_method_regression_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        desc 'delete with model, code 200',
+             is_array: false,
+             success: { code: 200, model: Foo, message: 'deleted' }
+        delete '/deleteme200' do
+          present({ something: 'bar' }, with: Foo)
+        end
+
+        desc 'delete with model, code 204',
+             is_array: false,
+             success: { code: 204, model: Foo, message: 'deleted' }
```
Comment (LeFnord) [inline spec/issues/591_delete_method_regression_spec.rb]: the default status code for delete is 204, with an empty body, but here you are suggesting to respond an entity `Foo` for an empty body 204

Comment (mattyr) [inline spec/issues/591_delete_method_regression_spec.rb]: Correct.  The existing code (prior to this change) modified the 204 to be a 200 if a response body (model) was given.  I maintained that behavior here (ie, I'm not introducing something new here).

File: spec/issues/591_delete_method_regression_spec.rb (hunk 1 above)
Comment (mattyr) [inline spec/issues/591_delete_method_regression_spec.rb]: it's another attempt to maintain the existing behavior.  technically it's a bad definition, but i didn't want to introduce a brand new regression.  in this scenario the "correct" response definition is appended.  if it's safe to modify this behavior, i can do so.

File: lib/grape-swagger/endpoint.rb [hunk 3]
```diff
@@ -198,7 +198,7 @@ def response_object(route)
         response_model = expose_params_from_model(value[:model]) if value[:model]
 
         if route.request_method == 'DELETE' && !value[:model].nil?
```
Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: habe this change (https://github.com/ruby-grape/grape-swagger/pull/583/files#diff-7af39528837333c960830618fb6a52e9L192) must be reverted

Comment (mattyr) [inline lib/grape-swagger/endpoint.rb]: Sorry, I'm not sure what you mean, do you want me to revert that change?  That's part of the change that introduced the regression.  But I think that altered behavior intentionally, correct?  If not, then I can revert that change, and I think my new unit tests will still pass; however that means that the new behavior introduced by that line will be reverted back to what it was... is that acceptable?  (sorry if i'm misunderstanding you!)

**Comment by mattyr** [comment]
rebased

File: lib/grape-swagger/endpoint.rb (hunk 3 above)
Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: yeap, think for the moment – support grape < 0.19.0 – it is the best, the default of 204 for DELETE was introduced in 0.19.0

Comment (mattyr) [inline lib/grape-swagger/endpoint.rb]: sorry, i'm still confused about how you'd like to address the problem.  i'm willing to make alterations to this pull request.  do you want to:

a) Assert compatibility via a dependency lock on < 0.19.0, and make the changes specific to that version
b) Revert the change that I mention above and restore compatibility with all versions
c) Something else?

Sorry for being confused...

**Comment by LeFnord** [comment]
@mattyr … sorry for the delay, will revert the changes I mentioned above

**Comment by LeFnord** [comment]
@mattyr please can you check if it also happen with this PR https://github.com/ruby-grape/grape-swagger/pull/597

**Comment by mattyr** [comment]
looks like #597 does fix it.  will close this pull request.
